txscript/engine.go

```diff
@@ -555,6 +560,13 @@ func (vm *Engine) checkHashTypeEncoding(hashType SigHashType) error {
 // checkPubKeyEncoding returns whether or not the passed public key adheres to
 // the strict encoding requirements if enabled.
 func (vm *Engine) checkPubKeyEncoding(pubKey []byte) error {
+	if vm.hasFlag(ScriptVerifyWitnessPubKeyType) && vm.witness &&
```

Comment:
That's what the PR does in earlier commits.  It uses `vm.witness` to specifically mean that it is a v0 witness program as opposed to just a witness program.